src/media-groups.js

```diff
@@ -138,12 +138,23 @@ export const onTrackChanged = (type, settings) => () => {
     return;
   }
 
-  if (!activeGroup.playlistLoader) {
-    // when switching from demuxed audio/video to muxed audio/video (noted by no playlist
-    // loader for the audio group), we want to do a destructive reset of the main segment
-    // loader and not restart the audio loaders
-    mainSegmentLoader.resetEverything();
-    return;
+  if (type === 'AUDIO') {
```

Comment:
I think this just handles moving from demuxed audio/video to muxed audio/video in a better way.